src/Impl/Widget.cpp

```diff
@@ -0,0 +1,98 @@
+#include "Widget.h"
+
+#include "Plug/Event.h"
+#include "Plug/Graphics.h"
+#include "Plug/LayoutBox.h"
+
+Widget::Widget(const plug::LayoutBox& box) : box_(box.clone()) {}
+
+Widget::Widget(const Widget& widget) : box_(nullptr) { operator=(widget); }
+
+Widget& Widget::operator=(const Widget& widget)
+{
+  delete box_;
+  box_ = widget.box_->clone();
+  return *this;
+}
+
+Widget::~Widget() { delete box_; }
+
+void Widget::onEvent(const plug::Event& event, plug::EHC& context)
+{
+  switch (event.getType())
+  {
+  case plug::EventType::Tick:
+  {
+    onTick((plug::TickEvent&)event, context);
```

Comment:
I think `static_cast<T>` is the way to do this in C++. If you still don't want to use it, at least avoid casting away const-ness (cast to `const plug::TickEvent&` here).